iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp

```diff
@@ -302,39 +339,106 @@ struct TileAndDistributeOnTensorsPattern
                                tiledLinalgOp, rootOpAttr.getInt());
 
     tiledLinalgOp.op.getOperation()->removeAttr(kRootOpAttr);
+
+    // TODO(nicolasvasilache): return `clonedOp->getResults()` once we have
+    // shape operands and we drop tie_shape.
     SmallVector<Value, 4> shapedResults =
         createDispatchTieShapeOp(rewriter, linalgOp, dispatchOp);
     rewriter.replaceOp(op, shapedResults);
     return success();
   }
 };
 
-template <typename OpTy>
-static Value buildFlowWorkgroupInfoOp(OpBuilder &b, unsigned dim) {
-  return b.template create<OpTy>(b.getInsertionPoint()->getLoc(), dim);
-}
+static Optional<SmallVector<SmallVector<Value, 4>, 1>> computeOutputShape(
+    OpBuilder &builder, Operation *op) {
+  SmallVector<SmallVector<Value, 4>, 1> outputShapes;
+  for (auto outputType : op->getResultTypes()) {
+    // Add empty shape for scalar values.
+    if (outputType.isIntOrFloat()) {
+      outputShapes.push_back({});
+      continue;
+    }
 
-/// Returns true if an operation is to always be cloned into the dispatch region
-/// when its value is used within it.
-bool isAlwaysClonedIntoDispatchOp(Operation *op) {
-  if (isa<linalg::InitTensorOp, linalg::TensorReshapeOp>(op)) {
-    return true;
+    // TODO(ravishankarm): For now only handle static shapes. For dynamic
+    // shapes, the shape of the output needs to be resolved using tie shapes,
+    // etc.
+    if (auto shapedType = outputType.dyn_cast<ShapedType>()) {
+      if (!shapedType.hasStaticShape()) return llvm::None;
+      outputShapes.push_back(llvm::to_vector<4>(
+          llvm::map_range(shapedType.getShape(), [&](int64_t dim) -> Value {
+            return builder.create<ConstantIndexOp>(op->getLoc(), dim);
+          })));
+      continue;
+    }
+    return llvm::None;
   }
-  if (auto constantOp = dyn_cast<ConstantOp>(op)) {
-    return constantOp.getResult().getType().isIntOrFloat();
+  return outputShapes;
+}
+
+/// Rewrite pattern to put ops that are not-tilable or arent tiled. For example
```

Comment:
Puts stand-alone ops that are not tileable/tiled into its own `flow.dispatch_workgroups` op?